dropwizard-async-http-client/src/main/java/smartthings/dw/asynchttpclient/AsyncHttpClientModule.java

```diff
@@ -0,0 +1,36 @@
+package smartthings.dw.asynchttpclient;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Named;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+import org.asynchttpclient.config.AsyncHttpClientConfigDefaults;
+import org.asynchttpclient.config.AsyncHttpClientConfigHelper;
+import org.asynchttpclient.filter.RequestFilter;
+import java.util.Set;
+
+public class AsyncHttpClientModule extends AbstractModule {
+	@Override
+	protected void configure() {
+		Multibinder<RequestFilter> requestFilters = Multibinder.newSetBinder(binder(), RequestFilter.class);
+		requestFilters.addBinding().to(CorrelationIdFilter.class);
+	}
+
+	@Provides
+	@Singleton
+	AsyncHttpClient asyncHttpClient(AsyncHttpClientConfig config,  Set<RequestFilter> requestFilters) {
+		String prefix = AsyncHttpClientConfigDefaults.ASYNC_CLIENT_CONFIG_ROOT;
+		// Set override properties from config so we don't have to use a properties file
+		config.getProperties().forEach((name, value) -> System.setProperty(prefix + name, value.toString()));
```

Comment:
yeah, all the default values route through these methods to get the values:
https://github.com/AsyncHttpClient/async-http-client/blob/master/client/src/main/java/org/asynchttpclient/config/AsyncHttpClientConfigHelper.java#L65
you can create a new builder from that config and then set the properties explicitly but then you basically nee to implement a mapper from each key in the yaml to the setter on the builder.